pkg/fanal/analyzer/pkg/bottlerocket/bottlerocket.go

```diff
@@ -0,0 +1,109 @@
+package bottlerocket
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"slices"
+	"time"
+
+	"github.com/aquasecurity/trivy/pkg/fanal/analyzer"
+	"github.com/aquasecurity/trivy/pkg/fanal/types"
+	"github.com/aquasecurity/trivy/pkg/log"
+)
+
+func init() {
+	analyzer.RegisterAnalyzer(newBottlerocketPkgAnalyzer())
+}
+
+const analyzerVersion = 1
+
+var requiredFiles = []string{
+	"aarch64-bottlerocket-linux-gnu/sys-root/usr/share/bottlerocket/application-inventory.json",
+	"x86_64-bottlerocket-linux-gnu/sys-root/usr/share/bottlerocket/application-inventory.json",
+}
+
+type bottlerocketPkgAnalyzer struct{}
+
+func newBottlerocketPkgAnalyzer() *bottlerocketPkgAnalyzer { return &bottlerocketPkgAnalyzer{} }
+
+type ApplicationInventory struct {
+	Content []struct {
+		Name            string    `json:"Name"`
+		Publisher       string    `json:"Publisher"`
+		Version         string    `json:"Version"`
+		Release         string    `json:"Release"`
+		InstalledTime   time.Time `json:"InstalledTime"`
+		ApplicationType string    `json:"ApplicationType"`
+		Architecture    string    `json:"Architecture"`
+		URL             string    `json:"Url"`
+		Summary         string    `json:"Summary"`
+	} `json:"Content"`
+}
+
+func (a bottlerocketPkgAnalyzer) Analyze(ctx context.Context, input analyzer.AnalysisInput) (*analyzer.AnalysisResult, error) {
+	ctx = log.WithContextPrefix(ctx, "application-inventory")
+	parsedPkgs, err := a.parseApplicationInventory(ctx, input.Content)
+	if err != nil {
+		return nil, err
+	}
+
+	return &analyzer.AnalysisResult{
+		PackageInfos: []types.PackageInfo{
+			{
+				FilePath: input.FilePath,
+				Packages: parsedPkgs,
+			},
+		},
+	}, nil
+}
+
+func (a bottlerocketPkgAnalyzer) parseApplicationInventory(ctx context.Context, r io.Reader) ([]types.Package, error) {
+	var pkgs []types.Package
+
+	b, err := io.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
+
+	var applicationInventory ApplicationInventory
+	err = json.Unmarshal(b, &applicationInventory)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Is there any reason all the file content needs to be loaded?

```suggestion
	var applicationInventory ApplicationInventory
	if err := json.NewDecoder(r).Decode(&applicationInventory); err != nil {
		return nil, err
	}
```

Reply:
Done.